src/merkle_tree.rs

```diff
@@ -0,0 +1,1383 @@
+// This file contains the code used to create a full MerkleSet and is heavily reliant on the code in merkle_set.rs.
+
+use clvmr::sha2::{Digest, Sha256};
+use crate::merkle_set::{
+    hash,
+    BLANK,
+    get_bit,
+    NodeType,
+};
+#[cfg(test)]
```

Comment:
If you move the tests into a module, you wouldn't need these repeated `#[cfg(test)]` attributes. Given, this is something we do in other parts of the code as well - I'd like to clean 
them up eventually. Pretty minor but they get unruly fast if there's a lot of imports or helper functions.